apps/crn-server/src/data-providers/contentful/analytics.data-provider.ts

```diff
@@ -35,131 +117,48 @@ export class AnalyticsContentfulDataProvider implements AnalyticsDataProvider {
                 []),
             ]),
+            interestGroupMemberCount: team.inactiveSince
+              ? 0
+              : getUniqueIdCount([
+                  ...getCurrentInterestGroupIdsFromTeamLeaders(team),
+                  ...getCurrentInterestGroupIdsFromInterestGroupCollection(
+                    team,
+                  ),
+                ]),
             interestGroupPreviousMemberCount: getUniqueIdCount([
-              ...(team.linkedFrom?.teamMembershipCollection?.items.flatMap(
-                (teamMembershipItem) =>
-                  teamMembershipItem?.linkedFrom?.usersCollection?.items
-                    .flatMap(flattenInterestGroupLeaders)
-                    .filter(
-                      (item) =>
-                        team.inactiveSince ||
-                        item.userIsAlumni ||
-                        !item.interestGroupActive,
-                    )
-                    .map((item) => item.interestGroupId),
-              ) || []),
-              ...(team.linkedFrom?.interestGroupsCollection?.items
-                .filter(
-                  (interestGroup) =>
-                    team.inactiveSince || !interestGroup?.active,
-                )
-                .flatMap((interestGroup) => interestGroup?.sys.id) || []),
+              ...getPreviousInterestGroupIdsFromTeamLeaders(team),
+              ...getPreviousInterestGroupIdsFromInterestGroupCollection(team),
             ]),
             workingGroupLeadershipRoleCount: team.inactiveSince
               ? 0
               : getUniqueIdCount(
-                  team.linkedFrom?.teamMembershipCollection?.items.flatMap(
-                    (teamMembershipItem) =>
-                      teamMembershipItem?.linkedFrom?.usersCollection?.items
-                        .filter(filterOutAlumni)
-                        .flatMap(flattenWorkingGroupLeaders),
-                  ) || [],
-                ),
-            workingGroupMemberCount: team.inactiveSince
-              ? 0
-              : getUniqueIdCount(
-                  team.linkedFrom?.teamMembershipCollection?.items.flatMap(
-                    (teamMembershipItem) =>
-                      teamMembershipItem?.linkedFrom?.usersCollection?.items
-                        .filter(filterOutAlumni)
-                        .flatMap(flattenWorkingGroupMember),
-                  ) || [],
+                  getCurrentWorkingGroupIdsFromTeamLeaders(team),
                 ),
             workingGroupPreviousLeadershipRoleCount: getUniqueIdCount([
               ...((team.inactiveSince &&
-                team.linkedFrom?.teamMembershipCollection?.items.flatMap(
-                  (teamMembershipItem) =>
-                    teamMembershipItem?.linkedFrom?.usersCollection?.items
-                      .filter(filterOutAlumni)
-                      .flatMap(flattenWorkingGroupLeaders),
-                )) ||
+                getCurrentWorkingGroupIdsFromTeamMembers(team)) ||
```

Comment:
you would think that wouldn't you - I tried removing it but the issue is that if you use the `&&` expression then if its falsy it returns the value of the first element which is then spread in the array, and since the value of the `inactiveSince` is not iterable it throws an error, even if we do `!!inactiveSince` that's still the same case